woocommerce/wc-admin
# Remove the use of Dashicons and replace with `@wordpress/icons` or `gridicons`

Fixes #3613 

As suggested in #3613, this PR removes the use of Dashicons anywhere within wcAdmin and makes use of the `@wordpress/icons` or `gridicons`, as they are tree-shakeable.
This does actually make our bundle size a little bit bigger, as the icons are individually imported, and we used `window.wp.components` for dashicons.
~~*Note: I currently can't seem to get the wordpress/icons tree shaking to work, and it is pulling in the entire library.~~
It seems all good now.
cc'ing @senadir & @samueljseay incase either of you know what I am missing, this is what a [quick gif of the package](https://d.pr/i/55WkM8) compiled into the `app/index.js`. 

### Screenshots

### Detailed test instructions:

- Load this branch
- Check the updated icons, there are quite a few
    - The android banner app dismiss icons (I [uncommented this condition](https://github.com/woocommerce/woocommerce-admin/blob/main/client/mobile-banner/index.js#L44), and [removed display none](https://github.com/woocommerce/woocommerce-admin/blob/main/client/mobile-banner/style.scss#L27)) 
    - The move up/down and trash icons in the Dashboard overview (click on the ellipsis)
    - Remove a couple items and also click on the plus icon to add a view back in
    - Go to the products analytics page and select a custom date (the calendar icons should show)
    - Icon in shipping label I added a `true ||` for testing purposes[ here](https://github.com/woocommerce/woocommerce-admin/blob/main/src/Features/ShippingLabelBanner.php#L88)
